core/aws-core/src/test/java/software/amazon/awssdk/awscore/interceptor/FriendlyUnknownHostExceptionInterceptorTest.java

```diff
@@ -0,0 +1,107 @@
+        assertThat(modifyException(exception, Region.of("cn-north-99"), "millems-hotdog-stand"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericHelp_forServicesRegionalizedInAllPartitions() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.US_EAST_1, "dynamodb"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("network");
+    }
+
+    @Test
+    public void modifyException_returnsGenericGlobalRegionHelp_forServicesGlobalInSomePartitionOtherThanTheClientPartition() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.of("cn-north-99"), "iam"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("aws-global")
+            .hasMessageContaining("aws-cn-global");
+    }
+
+    @Test
+    public void modifyException_returnsSpecificGlobalRegionHelp_forServicesGlobalInTheClientRegionPartition() {
+        UnknownHostException exception = new UnknownHostException();
+        assertThat(modifyException(exception, Region.of("cn-north-1"), "iam"))
+            .isInstanceOf(SdkClientException.class)
+            .hasMessageContaining("aws-cn-global");
+    }
+
+    private Throwable modifyException(Throwable throwable) {
+        return modifyException(throwable, null);
+    }
+
+    private Throwable modifyException(Throwable throwable, Region clientRegion) {
+        return modifyException(throwable, clientRegion, null);
+    }
+
+    private Throwable modifyException(Throwable throwable, Region clientRegion, String serviceEndpointPrefix) {
+        SdkRequest sdkRequest = Mockito.mock(SdkRequest.class);
+        return INTERCEPTOR.modifyException(DefaultFailedExecutionContext.builder()
+                                                                        .interceptorContext(InterceptorContext.builder()
+                                                                                                              .request(sdkRequest)
+                                                                                                              .build())
+                                                                        .exception(throwable)
+                                                                        .build(),
+                                           new ExecutionAttributes().putAttribute(AwsExecutionAttribute.AWS_REGION,
+                                                                                  clientRegion)
+                                                                    .putAttribute(AwsExecutionAttribute.ENDPOINT_PREFIX,
+                                                                                  serviceEndpointPrefix));
```

Comment:
Can you indent these lines in a more readable way?

Reply:
I thought this was the most readable way :(

Reply:
I'll break up the call.

Reply:
:-). Usually the indentation rules are good but in this case it just ended up like a jumble to me